support/sier-codec/tests/integrated.rs

```diff
@@ -57,6 +57,30 @@ fn construct_numbers() {
     assert_eq!(message["bar"].as_u32(), Some(43));
 }
 
+#[test]
+fn serialize_object() {
+    let mut parser = Parser::default();
+    parser.add_file_defs(MULTIPLE_NUMBERS).unwrap();
+
+    let def = parser.struct_def("Foo").unwrap();
+    let message = def
+        .builder()
+        .set("foo", 42u64)
+        .set("bar", 43u32)
+        .try_build()
+        .unwrap();
+
+    let encoded = message.serialize();
+    let expected = def
+        .id()
+        .iter()
+        .chain(&[42, 0, 0, 0, 0, 0, 0, 0, 43, 0, 0, 0])
+        .cloned()
+        .collect::<Vec<_>>();
```

Comment:
I was thinking about this regarding the testing philosophy. Is there any way we can avoid computing the expected value? The reasoning is that if we're computing the expected, there is computational logic involved and that can be faulty and hideaway bugs. It also helps with clarity for the reader to know exactly the expected value. However, it's not always practical to give out a hard-coded value. So if it's not possible, this should be fine, but, in general, I'd prefer to have the expectation hard-coded and easier on the eye. What are your thoughts?

Reply:
The main reason we're doing this is to specify it starts with the ID. I think it'll be better to check different regions of the output as equal.

```
assert_eq!(&result[0..8], struct_def.id());
assert_eq!(&result[8..], &[42, 0, 0, 0]);
```

I'm not gonna do that in this PR. I'll probably do it next time I'm somewhere that is doing this type of work.